OpenRA.Game/Actor.cs

```diff
@@ -366,6 +379,18 @@ public bool IsTargetableBy(Actor byActor)
 			return false;
 		}
 
+		public WPos[] GetTargetablePositions()
```

Comment:
It might be nice if this was still a lazily generated enumerable - `Target.IsInRange` uses `Any` for example.

Reply:
I have no experience using Lazy, so I'm not sure what the exact code changes would need to look like.

Reply:
Sorry, I simply mean:
- Return `IEnumerable<WPos>`
- Remove the two `ToArray()`s inside this method.

If the caller needs to enumerate the lot anyway, it can do so itself. If it can bail out partway - we've saved some work since we didn't need to enumerate the lot in order to shove it into an array.

Reply:
Oh, ok.